mmdeploy/core/rewriters/rewriter_utils.py

```diff
@@ -56,6 +59,66 @@ def import_function(path: str) -> Tuple[Callable, Optional[type]]:
         return obj, None
 
 
+def collect_env(backend: Backend, ir: IR, **kwargs):
+    from mmdeploy.utils import get_backend_version, get_codebase_version
+    env = dict(backend=backend, ir=ir)
+    env['mmdeploy'] = mmdeploy.__version__
+    env.update(get_backend_version())
+    env.update(get_codebase_version())
+    env.update(kwargs)
+    return env
+
+
+class Checker(metaclass=ABCMeta):
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def check(self, env: Dict) -> bool:
+        pass
+
+
+class BackendChecker(Checker):
+
+    def __init__(self, required_backend: Backend):
+        super().__init__()
+        self.required_backend = required_backend
+
+    def check(self, env: Dict) -> bool:
+        return env['backend'] == self.required_backend
+
+
+class IRChecker(Checker):
+
+    def __init__(self, required_ir: Backend):
+        super().__init__()
+        self.required_ir = required_ir
+
+    def check(self, env: Dict) -> bool:
+        return env['ir'] == self.required_ir
+
+
+class LibVersionChecker(Checker):
+
+    def __init__(self, lib: str, min_version=None, max_version=None):
+        super().__init__()
+        self.lib = lib
+        self.min_version = min_version
+        self.max_version = max_version
+
+    def check(self, env: Dict) -> bool:
+        from packaging import version
+        valid = True
+        if self.min_version is not None:
```

Comment:
logic is not good enough. For instance, if min_version is not met and max_version is met, then final `valid=True` while actually the min_version is not met.

Reply:
Yes! The codes are wrong. I will fix them!